public/app/plugins/datasource/cloudwatch/tracking.ts

```diff
@@ -0,0 +1,92 @@
+import { DashboardLoadedEvent } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+
+import { isCloudWatchLogsQuery, isCloudWatchMetricsQuery } from './guards';
+import pluginJson from './plugin.json';
+import { CloudWatchMetricsQuery, CloudWatchQuery, MetricEditorMode, MetricQueryType } from './types';
+
+interface CloudWatchOnDashboardLoadedTrackingEvent {
+  grafana_version?: string;
+  dashboard_id?: string;
+  org_id?: number;
+
+  /* The number of CloudWatch logs queries present in the dashboard*/
+  logs_queries_count: number;
+
+  /* The number of CloudWatch metrics queries present in the dashboard*/
+  metrics_queries_count: number;
+
+  /* The number of queries using the "Search" mode. Should be measured in relation to metrics_queries_count */
+  metrics_search_count: number;
+
+  /* The number of search queries that are using the builder mode. Should be measured in relation to metrics_search_count */
+  metrics_search_builder_count: number;
+
+  /* The number of search queries that are using the code mode. Should be measured in relation to metrics_search_count */
+  metrics_search_code_count: number;
+
+  /* The number of search queries that have enabled the `match exact` toggle in the builder mode. Should be measured in relation to metrics_search_builder_count */
+  metrics_search_match_exact_count: number;
+
+  /* The number of queries using the "Query" mode (AKA Metric Insights). Should be measured in relation to metrics_queries_count */
+  metrics_query_count: number;
```

Comment:
`Mode` may add to the confusion here since it's used to describe if it's `Builder` or  `Code` mode (see [this](https://github.com/grafana/grafana/blob/cloudwatch/feature-tracking/public/app/plugins/datasource/cloudwatch/types.ts#L25-L28) enum). I don't think `query` should be pluralized since it refer to the name of the type. This is hard...Can the comment be improved to make it more clear?

Reply:
How about including in the comment something like following example for the three variables involved?
```
metrics_queries_count: 21 = metrics_query_count: 7 + metrics_search_count: 14
```
(or instead an individual test which asserts those three variables alone)

Reply:
Yes good idea, will add that to the comment!